api/base_controller.go

```diff
@@ -293,13 +377,42 @@ func (c *BaseController) PatchObject(r *web.Request) (*web.Response, error) {
 	labels, _, _ := query.ApplyLabelChangesToLabels(labelChanges, objFromDB.GetLabels())
 	objFromDB.SetLabels(labels)
 
-	object, err := c.repository.Update(ctx, objFromDB, labelChanges, criteria...)
+	operationFunc := func(ctx context.Context, repository storage.Repository) (types.Object, error) {
+		return repository.Update(ctx, objFromDB, labelChanges, criteria...)
+	}
+
+	isAsync := r.URL.Query().Get(QueryParamAsync)
+	if isAsync == "true" {
+		log.C(ctx).Debugf("Request will be executed asynchronously")
+		if err := c.checkAsyncSupport(); err != nil {
+			return nil, err
+		}
+
+		operation, err := c.buildOperation(ctx, c.repository, types.IN_PROGRESS, types.UPDATE, objFromDB.GetID(), log.CorrelationIDFromContext(ctx))
+		if err != nil {
+			return nil, err
+		}
+
+		operationID, err := c.scheduler.Schedule(operations.Job{
```

Comment:
.Schedule can accept ctx as first argument and internally only use the values and create a new contextWithtimeout. probably the whole Stateecoontext struct and reqctx field on job can be removed